lib/diag.c

```diff
@@ -744,7 +1095,33 @@ int switchtec_diag_port_eq_tx_table(struct switchtec_dev *dev, int port_id,
 }
 
 /**
- * @brief Get the equalization FS/LF
+ * @brief Get the far end TX equalization table
+ * @param[in]  dev	Switchtec device handle
+ * @param[in]  port_id	Physical port ID
+ * @param[out] res      Resulting port equalization table
+ *
+ * @return 0 on success, error code on failure
+ */
+int switchtec_diag_port_eq_tx_table(struct switchtec_dev *dev, int port_id,
+				    enum switchtec_diag_link link,
+				    struct switchtec_port_eq_table *res)
+{
+	int ret = -1;
+
+
+	if (switchtec_is_gen5(dev)) {
+		ret = switchtec_gen5_diag_port_eq_tx_table(dev, port_id,
+				link, res);
+	} else if (switchtec_is_gen4(dev)){
+		ret = switchtec_gen4_diag_port_eq_tx_table(dev, port_id,
+				link, res);
+	}
+
```

Comment:
Style nit: surrounding {} should not be present for single-statement following if/else.

I have noticed additional style issues in the code changes. Please review the code and address these concerns.